main/src/org/jetbrains/bsp/bazel/servers/JavaBspServer.java

```diff
@@ -0,0 +1,87 @@
+package org.jetbrains.bsp.bazel.servers;
+
+import ch.epfl.scala.bsp4j.BuildTargetIdentifier;
+import ch.epfl.scala.bsp4j.JavacOptionsItem;
+import ch.epfl.scala.bsp4j.JavacOptionsParams;
+import ch.epfl.scala.bsp4j.JavacOptionsResult;
+import com.google.common.base.Joiner;
+import com.google.common.collect.Lists;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.eclipse.lsp4j.jsonrpc.messages.Either;
+import org.eclipse.lsp4j.jsonrpc.messages.ResponseError;
+import org.jetbrains.bsp.bazel.ActionGraphParser;
+import org.jetbrains.bsp.bazel.Uri;
+import org.jetbrains.bsp.bazel.resolvers.ActionGraphResolver;
+import org.jetbrains.bsp.bazel.resolvers.TargetsResolver;
+import org.jetbrains.bsp.bazel.utils.MnemonicsUtils;
+
+// TODO: This class *should* implement a `JavaBuildServer` interface,
+// TODO: now `buildTargetJavacOptions` method returns a `Either<ResponseError, JavacOptionsResult>`
+// TODO: instead of a `CompletableFuture<JavacOptionsResult>` because of the `BazelBspServer`
+// TODO: command executing (`executeCommand`) implementation.
+public class JavaBspServer {
+
+  private final TargetsResolver targetsResolver;
+  private final ActionGraphResolver actionGraphResolver;
+
+  private final String javac;
+  private final String kotlinc;
+  private final String execRoot;
+
+  // TODO: too many arguments!!, constants will be moved
```

Comment:
constants - do you mean javac and kotlinc?

Reply:
yes